src/scripts/main.js

```diff
@@ -1,3 +1,280 @@
+button.innerText = 'Save to table';
+
+form.append(button);
+
+function saveDataToTable(data) {
+  const tr = document.createElement('tr');
+
+  tr.innerHTML = `
+    <td>${data.name}</td>
+    <td>${data.position}</td>
+    <td>${data.office}</td>
+    <td>${data.age}</td>
+    <td>$${Number(data.salary).toLocaleString('en-US')}</td>
+  `;
+
+  body.append(tr);
+};
+
+const pushNotification = (title, description, type) => {
+  const messageConteiner = document.createElement('div');
+  const messageTitle = document.createElement('h2');
+  const messageDescription = document.createElement('p');
+
+  document.body.append(messageConteiner);
+  messageConteiner.append(messageTitle);
+  messageConteiner.append(messageDescription);
+  messageConteiner.classList.add('notification', type);
+  messageConteiner.classList.add(`${type}`);
+  messageConteiner.dataset.qa = 'notification';
+  messageTitle.classList.add('title');
+  messageTitle.textContent = title;
+  messageDescription.textContent = description;
+
+  setTimeout(() => messageConteiner.remove(), 3000);
+};
+
+function checkValidation(data) {
+  if (!data.name.match(/^[a-zA-Z ]*$/g)) {
+    pushNotification('Invalid data',
+      'Only English allowed!',
+      'error');
+
+    return false;
+  }
+
+  if (data.name.length < 4) {
+    pushNotification('Invalid name',
+      'Name is too short, enter the correct one!',
+      'error');
+
+    return false;
+  }
+
+  if (data.age < 18 || data.age > 90) {
+    pushNotification('Invalid age',
+      'Enter the correct age!',
+      'error');
+
+    return false;
+  }
+
+  if (data.position === '' || data.salary === '') {
+    pushNotification('Missing data',
+      'Please, fill all fields!',
+      'error');
+
+    return false;
+  }
+
+  pushNotification('Great!',
+    'New employee successfully added!',
+    'success'
+  );
+
+  return true;
+};
+
+button.addEventListener('click', (e) => {
+  e.preventDefault();
+
+  const data = Object.fromEntries(new FormData(form).entries());
+
+  if (!checkValidation(data)) {
+    return;
+  }
+
+  saveDataToTable(data);
+  form.reset();
+});
+
+body.addEventListener('dblclick', (e) => {
+  const target = e.target;
+  const input = document.createElement('input');
+  const initialValue = target.innerText;
+
+  input.classList.add('cell-input');
+  input.type = 'text';
+  input.name = 'cellInput';
+  target.innerText = '';
+  target.append(input);
+  input.focus();
+
+  const isString = target.cellIndex === 0
+    || target.cellIndex === 1;
+  const isName = target.cellIndex === 0;
+  const isOffice = target.cellIndex === 2;
+  const isAge = target.cellIndex === 3;
+  const isSalary = target.cellIndex === 4;
+
+  function changeCellContent(value = initialValue) {
+    target.innerText = value;
+    input.remove();
+  }
+
+  function saveEditedField() {
+    if (!input.value) {
+      changeCellContent();
+
+      return;
+    }
+
+    if ((isAge || isSalary) && !input.value.match(/[0-9]/)) {
```

Comment:
You are doing validation in `checkValidation`, and don't run the current function if validation fails, no need to check all fields here again. So `saveEditedField` is redundant, you can run `changeCellContent` instead of it.

Reply:
I didn't understand how it is possible to do the same without calling the function saveEditedField

Reply:
I didn't make it up how rewrite it for both cases, in two cases different types of data and their validation and sooo I don't know how make it

Reply:
let`s try to figure it out in fe_chat, it can be faster)